tests/prompt/test_vc.py

```diff
@@ -0,0 +1,130 @@
+import os
+from pathlib import Path
+import shutil
+import subprocess as sp
+import tempfile
+from unittest.mock import Mock
+
+import pytest
+
+from xonsh.environ import Env
+from xonsh.prompt import vc
+from tools import DummyEnv
+
+
+# Xonsh interaction with version control systems.
+VC_BRANCH = {"git": {"master", "main"}, "hg": {"default"}}
+
+
+@pytest.fixture(params=VC_BRANCH.keys())
+def test_repo(request, tmpdir_factory):
```

Comment:
two notes here:

1. Removed the "module" scope, because if a certain test adds a file, I didn't want that to affect other tests
2. Using pytest's `tmpdir` functions here and elsewhere instead of tempfile since (I think?) they should better cleanup things after the test finishes